src/Routes/RosPage/RosPage.js

```diff
@@ -21,14 +24,53 @@ class RosPage extends React.Component {
 
     constructor(props) {
         super(props);
+
+        this.state = {
+            page: 1,
+            perPage: 5
+        };
+
+        this.onSetPage = this.onSetPage.bind(this);
+        this.onPerPageSelect = this.onPerPageSelect.bind(this);
     }
 
     async componentDidMount() {
         await window.insights.chrome.auth.getUser();
-        this.props.fetchSystems();
+        this.fetchWithParams();
+    }
+
+    onSetPage(event, page) {
+        const { perPage } = this.state;
+        const pagination = { page, perPage };
+        this.updatePagination(pagination);
+    }
+
+    onPerPageSelect(event, perPage) {
+        const page = 1;
+        const pagination = { page, perPage };
+        this.updatePagination(pagination);
+    }
+
+    updatePagination(pagination) {
+        this.setState({ page: pagination.page, perPage: pagination.perPage });
+        this.fetchWithParams({ page: pagination.page, perPage: pagination.perPage });
```

Comment:
If you are going to follow the same shape, no need to re-assign. And would call directly the `fetchSystems` function.
```suggestion
        this.setState(pagination);
        fetchSystems(pagination);
```